### ISSUE TITLE
[Helm] Allow single binary with Object Storage 

### ISSUE THREAD

**Issue by trevorwhitney** [grafana/loki#8185]
**Is your feature request related to a problem? Please describe.**
Currently, the Helm chart only allows two modes of operation, single binary with local filesystem storage and scalable with object storage. This is missing a set of use cases which are two small to merit the resources of a replicated scalable deployment (3x read and 3x write pods), but would still like replication (3x single binary pods).

**Describe the solution you'd like**
 A way that exposes this configuration without exposing the "deployment type". Ideally, a newcomer to Loki should not need to know about the difference between "single binary" and "scalable" deployment. Instead, they should provide information about the size of their environment, and their replication needs, and the chart should provide a "recommended" deployment.

At the very least, this deployment method should be possible with the helm chart, even if it requires overriding the whole config file.

**Describe alternatives you've considered**
An alternative was provided in https://github.com/grafana/loki/pull/7942, however it forces the operator to know about the different deployment types.

**Comment by djotanov** [comment]
Personally I think that the operator should know the difference between single binary and microservices deployment type, and that is already nicely documented in LOKI documentation: https://grafana.com/docs/loki/latest/fundamentals/architecture/deployment-modes/
Deployment type should be a conscious choice instead of imposed by this chart.
Hiding that in this chart is unnecessarily limiting this chart, and actually complicating things.

**Comment by trevorwhitney** [comment]
@djotanov I disagree. I don't see a reason to add an additional configuration value when we can infer the deployment model from existing configuration options. I think the less config there is the better. But please share your feedback on #8284 to encourage wider discussion. Thanks!

**Comment by silverlyra** [comment]
I’m in the process of upgrading a Loki 2.4 deployment (Helm chart 2.8) to Loki 2.7 (chart 4.4.2). I have been running in the single-binary mode, writing to S3, and was disappointed when saw in the docs for the new Helm chart that this is no longer supported.

So I was pleasantly surprised when I brought up the new Helm release, and it came up in single binary mode. 🤔

```
❯ kubectl -n loki get pods
NAME     READY   STATUS    RESTARTS   AGE
loki-0   2/2     Running   0          39m
```

I checked the S3 bucket and it’s definitely writing there.

It seems I accidentally found a workaround by using `storage_config` to set this up, as that’s how I did things in the old Helm chart:

```yaml
storage:
  type: aws

storage_config:
  aws:
    s3: "s3://<region>/<bucket>"
  boltdb_shipper:
    shared_store: s3
```

I think this works specifically because I’m using S3. I set `storage.type` so that the chart will [generate](https://github.com/grafana/loki/blob/9920a27f06b9a808bcd0d7323a55041bc343a8bc/production/helm/loki/values.yaml#L141-L149) a `schema_config` automatically, but the chart’s `isUsingObjectStorage` [helper](https://github.com/grafana/loki/blob/9920a27f06b9a808bcd0d7323a55041bc343a8bc/production/helm/loki/templates/_helpers.tpl#L470-L472) checks if `storage.type` is one of `gcs`, `s3`, or `azure`, and the value `schema_config` [expects](https://grafana.com/docs/loki/latest/configuration/#period_config) in `object_store` is `aws`, not `s3`. 😇

But it looks like someone using one of the different object stores could leave `storage.type` unset and provide a `schema_config` manually to get the same behavior.
